ui/src/app/components/users-table/users-table.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { UserService } from '../../services/user.service';
+import { Subject } from 'rxjs';
+import { AuthService } from '../../services/auth.service';
+
+@Component({
+  selector: 'tc-users-table',
+  templateUrl: './users-table.component.html',
+  styleUrls: ['./users-table.component.scss'],
+})
+export class UsersTableComponent implements OnInit, OnDestroy {
+  readonly users$ = this.userService.orderedUsers$;
+  readonly displayedColumns: string[] = ['id', 'username', 'name', 'role'];
+
+  private readonly unsubscribe$ = new Subject<void>();
```

Comment:
@mazanec202 

The usual purpose of this construct is unsubscribing from manual subscriptions (done via `Observable.subscribe()`) in component logic. E.g. in `court-detail.component.ts`:

```typescript
this.route.params.pipe(takeUntil(this.unsubscribe$)).subscribe(...);
```

The `takeUntil()` operator completes the observable chain (i.e. causes unsubscription) on the first value from `this.unsubscribe$`, which is done in `ngOnDestroy()`:

```typescript
ngOnDestroy(): void {
  this.unsubscribe$.next();
  this.unsubscribe$.complete();
}
```

When Angular destroys the view, it calls the `ngOnDestroy()` method. The `this.unsubscribe$.next()` call multicasts a sort-of "ping" (a `void`-type value) to all `takeUntil()` operators subscribed to it. That causes them to complete their respective observable chains and break the subscriptions. Without this, the subscriptions would "hang" after the component is destroyed.

In this component, no manual subscriptions making use of the `takeUntil()`-based unsubscription pattern are present. Therefore the `unsubscribe$` subject is redundant.

(Note: the `async` pipe, used to subscribe to observables in the template, "cleverly" unsubscribes on its own.)

Reply:
I wasn't aware of this, thanks!